python/paddle/fluid/tests/unittests/test_cumprod_op.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import numpy as np
+
+from op_test import OpTest
+import random
+import paddle
+
+import paddle.nn as nn
+import paddle.nn.functional as F
+import paddle.fluid as fluid
+import paddle.fluid.core as core
+from paddle.fluid import compiler, Program, program_guard
+
+np.random.seed(0)
+
+
+# test function.
+class TestCumprod(OpTest):
+    def setUp(self):
+        paddle.enable_static()
+        self.op_type = "cumprod"
+        self.init_dtype()
+
+        x = (np.random.rand(2, 3, 10, 10) + 0.5).astype(self.dtype)
+        out = np.cumprod(x, axis=1)
+        self.inputs = {'X': x}
+        self.outputs = {'Out': out}
+        self.attrs = {'dim': 1}
+
+    def test_check_grad(self):
+        self.check_grad(['X'], 'Out')
+
+    def test_check_output(self):
+        self.check_output()
+
+    def init_dtype(self):
+        self.dtype = np.float64
+
+
+# test api.
+class TestCumprodAPI(unittest.TestCase):
+    def init_dtype(self):
+        self.dtype = 'float64'
+        self.shape = [2, 3, 10, 10]
+
+    def setUp(self):
+        paddle.enable_static()
+        self.init_dtype()
+        self.x = (np.random.rand(2, 3, 10, 10) + 0.5).astype(self.dtype)
+        self.place = [paddle.CPUPlace()]
+        if core.is_compiled_with_cuda():
+            self.place.append(paddle.CUDAPlace(0))
+
+    # test static graph api.
+    def test_static_api(self):
+        paddle.enable_static()
+
+        def run(place):
+            with paddle.static.program_guard(paddle.static.Program()):
+                X = paddle.fluid.data('X', self.shape, dtype=self.dtype)
```

Comment:
`X = ...` -> `x = ...`. Try to obey the conventional naming rules of Python when writing the Python codes.